### PR TITLE
Remove connection-specific headers from HTTP/2 message.

### PR DESCRIPTION
**Pull request by luborco**

### CODE REVIEW

File: pkg/transport/transport.go [hunk 1]
```diff
@@ -108,16 +108,16 @@ func createRoutingEndpoint(rc *Context, logger Logger) func(w http.ResponseWrite
...
 			}
 			logger.ErrorContext(r.Context(), jErrors.Details(jErrors.Trace(err)))
 			return
 		}
 
-		transformer.SetRESTHeaders(w.Header(), header, trailer)
-		fmt.Fprint(w, protojson.Format(rpcResponse))
+		transformer.SetRESTHeaders(r.ProtoMajor, w.Header(), header, trailer)
 		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, protojson.Format(rpcResponse))
```
Comment (MartinKuzma) [inline pkg/transport/transport.go]: Don't use Fprint for copying data to writer. io.copy or w.Write are better options.

File: pkg/service/transformer/transformer.go [hunk 2]
```diff
@@ -17,42 +17,71 @@ import (
...
-func GetRPCRequestContext(request *http.Request) context.Context {
+func isConnectionSpecificHeader(name string) bool {
+	switch name {
+	case headerConnection, headerProxyConnection, headerKeepAlive, headerTransferEncoding, headerUpgrade:
+		return true
+	}
+	return false
+}
+
+func GetRPCRequestContext(protoMajor int, request *http.Request) context.Context {
```
Comment (MartinKuzma) [inline pkg/service/transformer/transformer.go]: Doesn't require protoMajor argument. Protomajor is already present in request.

File: pkg/service/transformer/transformer.go [hunk 3]
```diff
@@ -17,42 +17,71 @@ import (
...
+	grpcMetadata.Set(headerContentType, "application/protobuf")
 
 	return metadata.NewOutgoingContext(request.Context(), grpcMetadata)
 }
 
-func SetRESTHeaders(headers http.Header, gRPCheader metadata.MD, gRPCTrailer metadata.MD) {
+func setHeader(headers http.Header, protoMajor int, name string, values []string) {
+	for _, value := range values {
+		// RFC 9113 8.2.2.: endpoint MUST NOT generate an HTTP/2 message containing connection-specific header fields
+		if protoMajor > 1 && isConnectionSpecificHeader(name) {
```
Comment (MartinKuzma) [inline pkg/service/transformer/transformer.go]: This is incorrect place for checking name. For each value you are doing same name check. Instead, move this check above for loop.

**Review by MartinKuzma — CHANGES_REQUESTED** [review]

File: pkg/service/transformer/transformer.go [hunk 4]
```diff
@@ -17,42 +17,71 @@ import (
...
-	grpcMetadata.Set("accept", "application/protobuf")
-	grpcMetadata.Set("content-type", "application/protobuf")
+	grpcMetadata.Set(headerAccept, "application/protobuf")
+	grpcMetadata.Set(headerContentType, "application/protobuf")
 
 	return metadata.NewOutgoingContext(request.Context(), grpcMetadata)
 }
 
-func SetRESTHeaders(headers http.Header, gRPCheader metadata.MD, gRPCTrailer metadata.MD) {
+func setHeader(headers http.Header, protoMajor int, name string, values []string) {
```
Comment (danielkraic) [inline pkg/service/transformer/transformer.go:64]: Optional: Move this function to the bottom of the file.

It is not required in Go to define a function before this function is used. Best practice for more readable code is to define exported functions first and then un-exported functions. Files are meant to be read from top to bottom, not the opposite way. Therefore the more important functions (exproted) should be defined before less important functions.

**Review by MartinKuzma — APPROVED** [review]

**Pull request merged**